service/activity/activity_handler.go

```diff
@@ -0,0 +1,24 @@
+package activity
```

Comment:
I love how you organized the project in packages except for inside the service package.
IMO it is overkill here. The separation between routes of types `books`, `book_search` and `store` for example.
I would do one `handlers.go` file for all routes, and also have a file for ES logic and one for Redis logic. (not including the middleware package which is great).
You don't have to do it like that, but if not try to skip the dto files because it seems a bit much now.

Also maybe I would call the `context_helper` package with something more generic like `utils`.